src/components/RouteCard.tsx

```diff
@@ -1,20 +1,93 @@
 
 import Avatar from '~/components/material/Avatar'
-import Card, { CardContent, CardHeader } from '~/components/material/Card'
+import { CardContent, CardHeader } from '~/components/material/Card'
 import Icon from '~/components/material/Icon'
+import RouteOptions from '~/components/RouteOptions'
 import RouteStaticMap from '~/components/RouteStaticMap'
 import RouteStatistics from '~/components/RouteStatistics'
+import Timeline from './Timeline'
 
-import type { RouteSegments } from '~/types'
+import type { Route, RouteSegments } from '~/types'
 
-const RouteHeader = (props: { route: RouteSegments }) => {
-  const startTime = () => dayjs(props.route.segment_start_times[0])
-  const endTime = () => dayjs(props.route.segment_end_times.at(-1))
+import { reverseGeocode } from '~/map'
 
-  const headline = () => startTime().format('ddd, MMM D, YYYY')
-  const subhead = () => `${startTime().format('h:mm A')} to ${endTime().format('h:mm A')}`
+const [showRouteOptionsCard, setShowRouteOptionsCard] = createSignal(false)
+const [currentCardRoute, setCurrentCardRoute] = createSignal<Route | undefined>(undefined)
+
+const RouteOptionsCard: Component = () => {
+  const [isMdOrLarger, setIsMdOrLarger] = createSignal(false)
+
+  // listen isMdOrLarger
+  createEffect(() => {
+    const updateSize = () => {
+      setIsMdOrLarger(window.innerWidth >= 768)
+    }
+    window.addEventListener('resize', updateSize)
+    // Initial check
+    updateSize() 
+
+    return () => window.removeEventListener('resize', updateSize)
+  })
+
+  const stopPropagation = (event: MouseEvent) => {
+    event.stopPropagation()
+
+    setShowRouteOptionsCard(false)
+  }
+
+  return (
+    <Show when={showRouteOptionsCard()}>
+      <div onClick={stopPropagation} classList={{
+        'fixed inset-0 z-50 flex items-center justify-center bg-black bg-opacity-50 transition duration-800 ease-in-out': isMdOrLarger(),
+        'fixed inset-0 z-50 flex flex-col-reverse bg-black bg-opacity-50 p-4 transition duration-800 ease-in-out': !isMdOrLarger(),
+        'transform translate-y-0': !isMdOrLarger() && showRouteOptionsCard(),
+        'transform translate-y-full': !isMdOrLarger() && !showRouteOptionsCard(),
+      }}>
+        <div onClick={(event: MouseEvent) => event.stopPropagation()}>
+          <RouteOptions route={currentCardRoute()} />
+        </div>
+      </div>
+    </Show>
+  )
+}
+
+type FavoriteRoutes = string[]
+
+const RouteHeader = (props: { route?: RouteSegments }) => {
+
+  const startTime = () => props?.route?.segment_start_times ? dayjs(props.route.segment_start_times[0]) : null
+  const endTime = () => props?.route?.segment_end_times ? dayjs(props.route.segment_end_times.at(-1)) : null
+
+  const headline = () => startTime()?.format('ddd, MMM D, YYYY')
+  const subhead = () => `${startTime()?.format('h:mm A')} to ${endTime()?.format('h:mm A')}`
+
+  const [isFavorite, setIsFavorite] = createSignal(false)
+
+  const toggleFavorite = (routeName: string) => {
```

Comment:
New features like favoriting should be in a seperate PR

Reply:
fixed